cmd/carbon-relay-ng/carbon-relay-ng.go

```diff
@@ -184,10 +184,10 @@ func main() {
 		_, err = statsmt.NewProcessReporter()
 		if err != nil {
 			//ProcessReporter depends on /proc which doesn't exist on OSX/Windows
-			if strings.HasSuffix(err.Error(), "stat /proc: no such file or directory") {
-				log.Warn("stats: could not initialize process reporter because /proc does not exist")
-			} else {
+			if runtime.GOOS == "linux" {
 				log.Fatalf("stats: could not initialize process reporter: %v", err)
```

Comment:
you want to keep the Fatalf in the linux case?
also why don't we just move the GOOS check around `statsmt.NewProcessReporter()` ? (not even try to create the reporter if it's not linux)

Reply:
Get rid of the `statsmt.NewProcessReporter` on non-linux platform was my first choice, I thought there was an agreement to keep it there, this code is from a previous or that tried to address the same issue....

Why it shouldn't be a `Fatalf` in Linux?

Reply:
> Why it shouldn't be a Fatalf in Linux?

oops i misread. it makes sense

> Get rid of the statsmt.NewProcessReporter on non-linux platform was my first choice, I thought there was an agreement to keep it there, this code is from a previous or that tried to address the same issue....

I think what we have right now is very abstract
either we do it in the error case, and change our handling based on what the error said,
or we say "this feature only works on this particular GOOS values" and hence it should only be enabled for those.
not a mix of both.

I actually like the error handling approach a bit more. perhaps there's other GOOS values for which /proc may exist? or people might do weird thing with emulation layers etc.  You said the error message is different on freeBSD so i would either maintain a list of recognized error messages, or perhaps the recent error improvements shipped in Go allow us to properly check the type of the error.

Reply:
I have different behaviour for the error...
* On FreeBSD, it fails earlier and `os.IsNotExist(err)` returns `true`
* On Mac OSX, it fails later on and `os.IsNotExist(err)` is `false`

I updated the pr with a check if the file `/proc/PID` exists as soon as possible. 
It is not reliable to use the error message to identify the error type, and the error returned by `procfs.NewProc` is not wrapped so there is no way (from my understanding of Golang) to understand what it is...